UOP1_Project/Assets/Scripts/AudioSystem.cs

```diff
@@ -0,0 +1,175 @@
+﻿using System;
+using UnityEngine;
+using System.Collections;
+using System.Collections.Generic;
+using Random = UnityEngine.Random;
+
+public class AudioSystem : MonoBehaviour
+{
+
+	private bool _sfxMute;
+	private bool _musicMute;
+	private AudioSource _theme;
+	private AudioSource _sfx;
+	private static readonly List<string> mixBuffer = new List<string>();
+	private const float mixBufferClearDelay = 0.05f;
+
+	internal string currentTrack;
+
+	public float delayInCrossfading = 0.3f;
+	public List<Theme> tracks = new List<Theme>();
+	public List<SFX> sounds = new List<SFX>();
+
+	private static int BoolToBinary(bool b) => b ? 1 : 0;
+	private SFX GetSoundByName(string sName) => sounds.Find(x => x.name == sName);
+
+	public float MusicVolume => !PlayerPrefs.HasKey("MusicVolume") ? 1f : PlayerPrefs.GetFloat("MusicVolume");
+	public float SfxVolume => !PlayerPrefs.HasKey("SFXVolume") ? 1f : PlayerPrefs.GetFloat("SFXVolume");
+
+	private void Awake()
+	{
+		// Configuring Audio Source For Playing Music And SFX
+		_theme = gameObject.AddComponent<AudioSource>();
+		_theme.loop = true;
+		_sfx = gameObject.AddComponent<AudioSource>();
+		_sfxMute = false;
+		_musicMute = false;
+		// Check If Sfx Volume Is Not 0
+		if (Math.Abs(SfxVolume) > 0.05f)
+			_sfx.volume = SfxVolume;
+		// Set The Values To 0
+		else
+			_sfx.volume = 0;
+		// Check If Music Volume Is Not 0
+		if (Math.Abs(MusicVolume) > 0.05f)
+			_theme.volume = MusicVolume;
+		// Set The Values To 0
+		else
+			_theme.volume = 0;
+		// Checks If The sfxMute Is True Or Not
+		if (PlayerPrefs.GetInt("sfxMute") == 1)
+		{
+			SfxToggle();
+		}
+		// Checks If The musicMute Is True Or Not
+		if (PlayerPrefs.GetInt("musicMute") == 1)
+		{
+			MusicToggle();
+		}
+
+		StartCoroutine(MixBufferRoutine());
+	}
+
+	// Responsible for limiting the frequency of playing sounds
+	private IEnumerator MixBufferRoutine()
+	{
+		float time = 0;
+
+		while (true)
+		{
+			time += Time.unscaledDeltaTime;
+			yield return 0;
+			if (time >= mixBufferClearDelay)
+			{
+				mixBuffer.Clear();
+				time = 0;
+			}
+		}
+	}
```

Comment:
Do I understand it correctly that this is for limiting the ability to play the same SFX twice in under 0.05?
I'm not sure we really need it, since we won't have that fast pace of random SFX, but at the same time we might have the same SFX play very close twice, and we want to hear both. Case in point: 2 same SFX happen but one is close and the other is far. I am supposed to hear both, or maybe just one, but for sure we want to hear the one close. With this method, the close one might be cutoff just because farther away, a similar SFX is playing.

Reply:
Unless I misunderstood the logic, in which case, apologies!